src/BizHawk.Client.Common/lua/LuaHelper.cs

```diff
@@ -9,5 +10,37 @@ public static class LuaExtensions
 	{
 		public static LuaTable EnumerateToLuaTable<T>(this NLuaTableHelper tableHelper, IEnumerable<T> list, int indexFrom = 1)
 			=> tableHelper.ListToTable(list as IReadOnlyList<T> ?? list.ToList(), indexFrom);
+
+		public static string Serialize(this LuaTable lti)
+		{
+			var sorted = lti
+				.OrderBy(static item => item.Key switch
+				{
+					long => 0,
+					string => 1,
+					double => 2,

```

Comment:
Can you add another test case with a mix of integers and floats? Because I have a feeling this will give an unintuitive ordering (which is part of why I only did `long`, that approximated Lua's sequence rules).

Reply:
Yeah, mixed integers and floats will not be sorted together. I don't think there's any simple solution to this without making the sorting much more complex, but maybe `double` should come after `long`.

I didn't include them in the test because their formatting is locale dependent (`.` vs `,`). Maybe that should change? FWIW, Lua's `string.format` seems to always use `.` as the decimal separator.